File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/utils/BuildUtil.java
package com.lilanz.tooldemo.utils;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.util.Log;

/**
 * 打包相关方法
 */
public class BuildUtil {


    /**
     * 获得应用版本名
     */
    public static String getVersionName(final Context con) {
        return getPackgeInfo(con).versionName;
    }

    /**
     * 获得应用版本号
     */
    public static int getVersionCode(final Context con) {
        return getPackgeInfo(con).versionCode;
    }

    /**
     * 获得应用包名
     */
    public static String getPackageName(final Context con) {
        return getPackgeInfo(con).packageName;
    }

    /**
     * 判断当前应用是否是debug状态
     */
    public static boolean isApkInDebug(Context context) {
        try {
            ApplicationInfo info = context.getApplicationInfo();
            return (info.flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
        } catch (Exception e) {
            return false;
        }
    }

    /**
     * 获取应用包相关信息
     */
    public static PackageInfo getPackgeInfo(final Context con) {
        PackageManager packageManager = con.getPackageManager();
        PackageInfo packageInfo = null;
        try {
            packageInfo = packageManager.getPackageInfo(con.getPackageName(), 0);
        } catch (PackageManager.NameNotFoundException e) {
            Log.e("BuildUtil", e.toString());
        }
        return packageInfo;
    }

}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/utils/BuildUtil.kt
package com.lilanz.kotlintool.utils

import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import java.lang.Exception

class BuildUtil {
    companion object {
        /**
         * 获取应用版本号
         */
        fun getVersionCode(context: Context): Int {
            var manager: PackageManager = context.packageManager
            try {
                var info: PackageInfo = manager.getPackageInfo(context.packageName, 0)
                return info.versionCode
            } catch (e: Exception) {
                e.printStackTrace()
            }
            return 0
        }

        /**
         * 判断应用是否处于debug
         */
        fun isInDebug(context: Context): Boolean {
            try {
                var info: ApplicationInfo = context.applicationInfo
                return (info.flags and ApplicationInfo.FLAG_DEBUGGABLE) != 0
            } catch (e: Exception) {
                return false
            }
        }
    }
}